syntax = "proto3";
package raid_fs;

import "error.proto";

/*
   RAID provides block IO API to the above filesystem.
*/

message BlockReadRequest {
  uint64 block_no = 1;
}

message BlockReadOKReply {
  bytes block = 1;
}

message BlockReadReply {
  oneof test_oneof {
    Error error = 1;
    BlockReadOKReply ok = 2;
  }
}

message BlockWriteRequest {
  uint64 block_no = 1;
  bytes block = 2;
}

message BlockWriteReply {
  optional Error error = 1;
}

service RAIDNode {
  rpc Read(BlockReadRequest) returns (BlockReadReply) {}
  rpc Write(BlockWriteRequest) returns (BlockWriteReply) {}
}
